chain/neutrino.go

```diff
@@ -358,17 +359,30 @@ func (s *NeutrinoClient) Rescan(startHash *chainhash.Hash, addrs []btcutil.Addre
 	// notification indicating the rescan has "finished".
 	if header.BlockHash() == *startHash {
 		s.finished = true
+		rescanQuit := s.rescanQuit
+
+		// Release the lock while dispatching the notification since
+		// it's possible for the notificationHandler to be waiting to
+		// acquire it before receiving the notification.
+		s.clientMtx.Unlock()
 		select {
 		case s.enqueueNotification <- &RescanFinished{
 			Hash:   startHash,
 			Height: int32(bestBlock.Height),
 			Time:   header.Timestamp,
 		}:
 		case <-s.quit:
+			// We'll need to re-acquire the lock in order for the
+			// deferred release call to not panic.
+			s.clientMtx.Lock()
 			return nil
-		case <-s.rescanQuit:
+		case <-rescanQuit:
+			// We'll need to re-acquire the lock in order for the
+			// deferred release call to not panic.
+			s.clientMtx.Lock()
 			return nil
 		}
+		s.clientMtx.Lock()
```

Comment:
I noticed that we are not returning the error on line 391

Reply:
Heh good catch!